src/lib/storage/segment_access_counter.hpp

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <atomic>
+#include <map>
+#include <memory>
+#include <string>
+
+#include "storage/pos_list.hpp"
+#include "types.hpp"
+
+namespace opossum {
+class Table;
+
+class SegmentAccessCounter {
+  friend class SegmentAccessCounterTest;
+
+ public:
+  template <typename T>
+  class Counter {
+   public:
+    T other;
+    T iterator_create;
+    T iterator_seq_access;
+    T iterator_increasing_access;
+    T iterator_random_access;
+    T accessor_create;
+    T accessor_access;
+    T dictionary_access;
+
+    Counter();
+
+    void reset();
+
+    T sum() const;
+
+    std::string to_string() const;
+
+    inline static const std::string HEADERS =
+        "Other,IteratorCreate,IteratorSeqAccess,IteratorIncreasingAccess,"
+        "IteratorRandomAccess,AccessorCreate,AccessorAccess,"
+        "DictionaryAccess";
+  };
```

Comment:
If you use the enum, you could create a string mapping and ensure that the order is correct and complete.